RealmTasks Android/app/src/main/java/io/realm/realmtasks/TaskActivity.java

```diff
@@ -184,21 +181,16 @@ public boolean onClicked(ItemViewHolder viewHolder) {
         @Override
         public void onChanged(ItemViewHolder viewHolder) {
             adapter.onItemChanged(viewHolder);
-            adapter.notifyItemChanged(viewHolder.getAdapterPosition());
         }
 
         @Override
         public void onAdded() {
             adapter.onItemAdded();
-            adapter.notifyItemInserted(0);
```

Comment:
Yeah you only need to add `notifyDataSetChanged()` when pulling to right (because it stays green or something) but otherwise the inserted/removed calls should be managed entirely by the change listener